src/config/schema-type.ts

```diff
@@ -211,6 +211,10 @@ export const schemaTypes = {
 			apiTimeout: PermissiveNumber,
 			registered_at: t.union([PermissiveNumber, NullOrUndefined]),
 			deviceId: t.union([PermissiveNumber, NullOrUndefined]),
+			supervisorVersion: t.union([t.string, t.undefined]),
+			osVersion: t.union([t.string, t.undefined]),
+			osVariant: t.union([t.string, t.undefined]),
+			macAddress: t.union([t.string, t.undefined]),
```

Comment:
```suggestion
			supervisorVersion: t.union([t.string, NullOrUndefined]),
			osVersion: t.union([t.string, NullOrUndefined]),
			osVariant: t.union([t.string, NullOrUndefined]),
			macAddress: t.union([t.string, NullOrUndefined]),
```

Or does the balena API not accept null for these fields?

Reply:
balena-device-register types don't accept null for them because it makes no sense to send null as they'll default to null, but the api would happily accept null as a reaffirmation of the default, it'd just be a small waste of bandwidth

Reply:
I see, that makes sense. Thanks for clarifying!